py/types.py

```diff
@@ -0,0 +1,16 @@
+from collections.abc import Sequence, Mapping
+from typing import TypedDict, Protocol
+
+
+class AIProvider(Protocol):
+
+  def __init__(self, config: Mapping[str, str|float|dict]) -> None:
+      pass
+
+  def request(self, messages: Sequence[Message]) -> List[str]:
```

Comment:
To support asynchronous streaming, it would need to return generator of chunks instead of a plain list

Reply:
Ok generator sounds good, will update it. I had a bit of trouble tracing how the generator is consumed.